### ISSUE TITLE
Upload in image type's context menu does not seem to work.

### ISSUE THREAD

**Issue by sookmax** [sanity-io/sanity#4096]
**Describe the bug**

When you have an image type field in your schema, you can upload an image from your local disk in a document pane. 

Uploading the initial image works fine, but when I try to change said uploaded image to something else through `Upload` button in the context menu (activated by the button with three dots at top-right corner), it doesn't seem to do anything -- no network activities, no processing indicators, and no warning/error messages.

**To Reproduce**

1. npm create sanity@latest
2. Create a project and select clean template
3. Add a schema with an image field
```ts
// ./schemas/defaultSchema.ts
import {defineField, defineType} from 'sanity'

export default defineType({
  type: 'document',
  name: 'defaultSchema',
  title: 'Default Schema',
  fields: [
    defineField({
      type: 'image',
      name: 'image',
      title: 'Image',
    }),
  ],
})

// ./schemas/index.ts
import defaultSchema from './defaultSchema'

export const schemaTypes = [defaultSchema]
```
5. Upload the initial image
6. Try changing the initial image with the `Upload` functionality provided by the context menu

**Expected behavior**

A newly selected image should be uploaded and replace the initial image.

**Screenshots**
![87A23FC4-4469-4D11-AE11-09B7BA662884](https://user-images.githubusercontent.com/71210554/212612217-00ed4a97-7316-4f2e-8fd2-6009e3141011.gif)

**Which versions of Sanity are you using?**

```
@sanity/cli (global)          3.2.3 (up to date)
@sanity/eslint-config-studio  2.0.1 (up to date)
@sanity/vision                3.2.3 (up to date)
sanity                        3.2.3 (up to date)
```

**What operating system are you using?**

`macOS Ventura 13.1 (Apple M2)`

**Which versions of Node.js / npm are you running?**

Run `npm -v && node -v` in the terminal and copy-paste the result here.

```
8.19.2
v18.12.1
```

**Comment by bjoerge** [comment]
Thanks for reporting @sookmax! We're aware and working on a fix.

**Comment by ifeltsweet** [comment]
Sorry to hijack this issue but a similar issue is happening with an array of images where when pressing "Upload" in the "Replace" image dialog is not doing anything. Also on Sanity v3.2.3.

![image](https://user-images.githubusercontent.com/1529466/212682237-11a516cc-2cc4-437d-9c6a-fed7402daa22.png)

**Comment by bjoerge** [comment]
@ifeltsweet No worries! It's the same root cause and will be fixed by #4094
